From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Wed, 19 Feb 2026 10:30:00 +0000
Subject: [PATCH 4/4] Allow oppo/oplus fingerprint HAL to find hwservice

The vendor.oppo and vendor.oplus fingerprint HIDL services are labeled
as default_android_hwservice since they have no explicit hwservice
context. The hal_fingerprint_oppo_compat domain needs permission to
find these services in the hwservice_manager, otherwise the fingerprint
HAL fails to register and the sensor becomes unresponsive.
---
 sepolicy/hal.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sepolicy/hal.te b/sepolicy/hal.te
--- a/sepolicy/hal.te
+++ b/sepolicy/hal.te
@@ -7,5 +7,9 @@ init_daemon_domain(hal_fingerprint_oppo_compat)
 
 
 type hal_fingerprint_oppo, domain;
+
+# Allow finding vendor oppo/oplus fingerprint HIDL services
+# which fall through to the default hwservice label.
+allow hal_fingerprint_oppo_compat default_android_hwservice:hwservice_manager find;
 allow hal_fingerprint_oppo vendor_default_prop:property_service { set };
 
